ipatests/test_integration/test_hsm.py

```diff
@@ -0,0 +1,1187 @@
+        )
+
+    def test_hsm_install_replica1_ipa_kra_install(self):
+        check_version(self.master)
+        tasks.install_kra(
+            self.replicas[1],
+            extra_args=('--token-password', self.token_password,)
+        )
+
+    def test_hsm_install_replica1_ipa_dns_install(self):
+        check_version(self.master)
+        tasks.install_dns(self.replicas[1])
+
+    def test_hsm_install_replica2_with_ca_kra_dns_install(self):
+        check_version(self.master)
+        tasks.install_replica(
+            self.master, self.replicas[2], setup_ca=True, setup_kra=True,
+            setup_dns=True,
+            extra_args=('--token-password', self.token_password,)
+        )
+
+    def test_hsm_install_master_ipa_kra_install(self):
+        check_version(self.master)
+        tasks.install_kra(
+            self.master,
+            extra_args=('--token-password', self.token_password,)
+        )
+
+    def test_hsm_install_client(self):
+        check_version(self.master)
+        tasks.install_client(self.master, self.clients[0])
+
+    def test_hsm_install_issue_user_cert(self):
+        check_version(self.master)
+        user = 'testuser1'
+        csr_file = f'{user}.csr'
+        key_file = f'{user}.key'
+        cert_file = f'{user}.crt'
+
+        tasks.kinit_admin(self.master)
+        tasks.user_add(self.master, user)
+        openssl_cmd = [
+            'openssl', 'req', '-newkey', 'rsa:2048', '-keyout', key_file,
+            '-nodes', '-out', csr_file, '-subj', '/CN=' + user]
+        self.master.run_command(openssl_cmd)
+
+        cmd_args = ['ipa', 'cert-request', '--principal', user,
+                    '--certificate-out', cert_file, csr_file]
+        self.master.run_command(cmd_args)
+
+    def test_hsm_install_healthcheck(self):
+        check_version(self.master)
+        set_excludes(self.master, "key", "DSCLE0004")
+        tasks.install_packages(self.master, ['*ipa-healthcheck'])
+        returncode, output = run_healthcheck(
+            self.master, output_type="human", failures_only=True
+        )
+        assert returncode == 0
+        assert output == "No issues found."
+
+
+class TestHSMInstallADTrustBase(BaseHSMTest):
+    """
+    Base test for builtin AD trust installation in combination with other
+    components with HSM support
+    """
+    num_replicas = 1
+    master_with_dns = False
+    master_with_kra = True
+
+    def test_hsm_adtrust_replica0_all_components(self):
+        check_version(self.master)
+        tasks.install_replica(
+            self.master, self.replicas[0], setup_ca=True,
+            setup_adtrust=True, setup_kra=True, setup_dns=True,
+            nameservers='master' if self.master_with_dns else None,
+            extra_args=('--token-password', self.token_password,)
+        )
+
+
+class TestADTrustInstallWithDNS_KRA_ADTrust(BaseHSMTest):
+
+    num_replicas = 1
+    master_with_dns = True
+    master_with_kra = True
```

Comment:
The test is missing master_with_ad = True

Reply:
Fixed. Added a temp commit running this test to be sure.